### PR TITLE
update iptables and add nftables

### PR DESCRIPTION
**Pull request by sknsean**

### CODE REVIEW

**Comment by kimrhh** [comment]
please try dropping the amversion hackery patch now that we have an updated buildhost

File: recipes/iptables/iptables.inc [hunk 1]
```diff
@@ -24,9 +18,6 @@ COMPATIBLE_HOST_ARCHS = ".*linux"
 
 inherit autotools pkgconfig auto-package-libs
 
-EXTRA_OECONF = "--disable-libnfnetlink \
-	      --disable-ipv6 \
```
Comment (kimrhh) [inline recipes/iptables/iptables.inc]: ipv6 should not be unconditional, please make it a useflag if you want it on

File: recipes/libmnl/libmnl_1.0.3.oe [hunk 2]
```diff
@@ -0,0 +1,19 @@
+SUMMARY = "Minimalistic user-space Netlink utility library"
+DESCRIPTION = "Minimalistic user-space library oriented to Netlink developers, providing \
+    functions for common tasks in parsing, validating, and constructing both the Netlink header and TLVs."
+
+LICENSE = "LGPLv2.1+"
```
Comment (kimrhh) [inline recipes/libmnl/libmnl_1.0.3.oe]: someday you might learn ;)

File: recipes/libnftnl/libnftnl_1.0.6.oe [hunk 3]
```diff
@@ -0,0 +1,14 @@
+SUMMARY = "Library for low-level interaction with nftables Netlink's API over libmnl"
+LICENSE = "GPLv2"
```
Comment (kimrhh) [inline recipes/libnftnl/libnftnl_1.0.6.oe]: tsk tsk

File: recipes/nftables/nftables_0.6.oe [hunk 4]
```diff
@@ -0,0 +1,20 @@
+SUMMARY = "Netfilter Tables userspace utillites"
+LICENSE = "GPLv2"
```
Comment (kimrhh) [inline recipes/nftables/nftables_0.6.oe]: one more

File: recipes/nftables/nftables/link-with-ncurses.patch [hunk 5]
```diff
@@ -0,0 +1,70 @@
+--- nftables-0.6/configure.ac.orig	2016-10-21 20:14:50.501289372 +0000
```
Comment (kimrhh) [inline recipes/nftables/nftables/link-with-ncurses.patch]: please add the patch header stuff here, I don't known why we need to link with ncurses :)

File: recipes/iptables/iptables.inc (hunk 1 above)
Comment (sknsean) [inline recipes/iptables/iptables.inc]: re-added

**Pull request merged**